proposals/dra-mode.md

```diff
@@ -0,0 +1,127 @@
+# Dynamic Resource Allocation plugin mode
+
+## Background
+
+Under the kubernetes SIG node, the Dynamic Resource Allocation plugin mechanism ([KEP #3063](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/3063-dynamic-resource-allocation#summary))
+aims to provide a new way to expose devices to kubernetes resources. While it doesn't aim to completely replace the Device Plugin system we currently
+use, it does overlap a lot with it.
+
+This new system is relevant to Akri as it would ease the management of slots usage and shared devices, for both instance level and configuration level resources.
+
+As a quick reference, the DRA uses a set of kubernetes resources to work: A ResourceClass that register a new class of devices that might be allocatable
+(with parameters for the driver) and a ResourceClaim that describe a request for allocation of a resource (with parameters for the driver).
+These resources all ultimately refer to a driver, that is a two components system comprised of a resource controller and a resource kubelet plugin.
+The controller will help the scheduler on node choice, and receive (de)allocation requests for the devices. The kubelet plugin will give the kubelet
+all needed information to expose the device to the workload for a given ResourceClaim. The tracking of available resources is completely up to the driver.
+
+In comparison, the current Akri model revolves around Configurations and Instances. A configuration both describe how to discover devices and what to
+schedule when they are discovered (broker logic). An instance just represent a discovered instance and the list of nodes where it is available.
+
+This proposal primarily focus on the Configuration resource, as the Instance resource is all about available resource tracking that is explicitly not
+covered by DRA.
+
+The Configuration resource match with the ResourceClass for the discovery part, the ResourceClaim part is handled by a combination of the Instance and
+inner workings of the akri agent.
+
+## API Changes
+
+These API changes may require writing a migration tool for existing Configurations
+
+### Create a `BrokerTemplate` resource kind
+
+In order to fit in the DRA model we need to separate the broker logic from the discovery logic, thus we need to create a new resource kind to handle the broker part. This resource needs to be linked to a discovery logic resource and provides all the needed information about what to schedule.
+
+This uses the "Deploy arbitrary Kubernetes resources as broker" for the description of what to schedule, please see that proposal for further details
+on the exact content of the broker fields.
+
+```yaml
+# apiVersion got ommited on purpose
+kind: BrokerTemplate
+metadata: ...
+spec:
+    discoveryReference: foo
+    instanceBrokerResources: ...
+    globalBrokerResources: ... # this field is named configurationBrokerResources in other proposal
+```
+
+### Create a `DiscoveryConfiguration` resource kind
```

Comment:
This is a cluster-scoped resource kind, as ResourceClasses are cluster-scoped. In fact while we currently have namespaces for our Configurations the DevicePlugin resource is expose outside this namespace, so it would makes more sense to have it cluster scoped anyway.